### PR TITLE
Update README.md

### LINKED ISSUE #37: Solution to Micro-Challenges
**MicroByte Title:** Array in Javascript

**Improvement Proposal:** Add solution to the micro-challenges 

**Benefits of proposed improvements:** To enable the readers understand different problem solving strategies

### PR DESCRIPTION
**Pull request by Sreejan-22**
## Closes #37

- Added approaches to solve micro-challenges given in the micro-byte
- Didn't give solutions right away. Instead mentioned the concept used or gave hints
- Added multiple approaches to solve the problems wherever possible
- Added new concepts on the topic.

### CODE REVIEW

**Review by Ak-Shaw — CHANGES_REQUESTED** [review]
@Sreejan-22 Kindly follow the contribution guidelines as given in CONTRIBUTING.md of the root directory of the repo. According to the contribution guidelines, you have to create an issue, before we know that you're working on something. This helps in maintaining proper workflow, also it helps us in further guiding you. Kindly use the `Update MicroByte` template for the issue.

Also, regarding your PR, kindly follow a `LearnByDoing` approach. You can use collapsible markdown as shown [here](https://gist.github.com/joyrexus/16041f2426450e73f5df9391f7f7ae5f). Kindly don't give away the solutions just like that, for we need the learners to get engages to task for their own good. 

I hope this gives you proper directions.

File: Array in Javascript/README.md [hunk 1]
```diff
@@ -76,6 +76,54 @@ The 4 element would be pushed to the end of the array. Notice that push returns
 Full code can be found at [here](src/insertion.js)
 ### Micro-Challenge
 Can you find A way to clone an array. Create a PR and add it here as Micro-Challenge Solution
```
Comment (Ak-Shaw) [inline Array in Javascript/README.md:78]: You need to remove the below statement.
`Create a PR and add it here as Micro-Challenge Solution.`

File: Array in Javascript/README.md [hunk 2]
```diff
@@ -108,6 +156,41 @@ No other element has been shifted, so it’s an O(1) runtime.
 Full code can be found at [here](src/deletion.js)
 ### Micro-Challenge
 How to remove multiple elements from array in JavaScript ? Give it a try create a PR and add it here as Micro-Challenge Solution
```
Comment (Ak-Shaw) [inline Array in Javascript/README.md:158]: Remove the following:
`create a PR and add it here as Micro-Challenge Solution`

File: Array in Javascript/README.md [hunk 3]
```diff
@@ -136,6 +219,37 @@ We would have to loop through the whole array (worst case) or until we find it:
 Full code can be found at [here](src/searching.js)
 ### Micro-Challenge
 Can you find more efficient solution ? Create a PR and add it here as Micro-Challenge solution
```
Comment (Ak-Shaw) [inline Array in Javascript/README.md:221]: Remove the following statement:
`Create a PR and add it here as Micro-Challenge solution`

**Review by Ak-Shaw — CHANGES_REQUESTED** [review]
@Sreejan-22 Kindly do the needful. All else seems fine.

**Review by Ak-Shaw — APPROVED** [review]
@Sreejan-22 Nicely done! :rocket:

Kindly wait till the review process ends. Till then you are free to explore other issues or create your own!

Cheers. :beers:

**Review by sudhanshutiwari264 — APPROVED** [review]
Looks good to me!✨
Nice work @Sreejan-22 !

**Comment by sudhanshutiwari264** [comment]
@all-contributors kindly add @Nitesh-thapliyal for:

- [x] doc

**Comment by sudhanshutiwari264** [comment]
@all-contributors kindly add @Sreejan-22  for:

- [x] doc

**Pull request merged**